source/2_platform/rex_windows/src/gui_application.cpp

```diff
@@ -142,6 +187,101 @@ namespace rex
+      }
+
+      void stop_resize(const event_system::Event& evt)
+      {
+        m_app_instance->resume();
+        m_app_state.remove_state(GuiApplicationState::Type::Resizing);
+
+        REX_ASSERT_X(evt.type == event_system::EventType::WindowStopWindowResize, "Event has to be of type \"WindowStopWindowResize\"");
+
+        resize(evt);
+      }
+
+      void minimize()
+      {
+        m_app_instance->pause();
+        m_app_state.add_state(GuiApplicationState::Type::Minimized);
+        m_app_state.remove_state(GuiApplicationState::Type::Maximized);
+      }
+
+      void maximize(const event_system::Event& evt)
+      {
+        m_app_instance->resume();
+        m_app_state.add_state(GuiApplicationState::Type::Maximized);
+        m_app_state.remove_state(GuiApplicationState::Type::Minimized);
+
+        REX_ASSERT_X(evt.type == event_system::EventType::WindowMaximized, "Event has to be of type \"WindowMaximized\"");
+
+        resize(evt);
+      }
+
+      void restore(const event_system::Event& evt)
+      {
+        REX_ASSERT_X(evt.type == event_system::EventType::WindowRestored, "Event has to be of type \"WindowRestored\"");
+
+        if(m_app_state.has_state(GuiApplicationState::Type::Minimized))
+        {
+          m_app_instance->resume();
+          m_app_state.remove_state(GuiApplicationState::Type::Minimized);
+
+          resize(evt);
+        }
+        else if(m_app_state.has_state(GuiApplicationState::Type::Maximized))
+        {
+          m_app_instance->resume();
+          m_app_state.remove_state(GuiApplicationState::Type::Maximized);
+
+          resize(evt);
+        }
+        else if(m_app_state.has_state(GuiApplicationState::Type::Resizing))
+        {
+          // If user is dragging the resize bars, we do not resize
+          // the buffers here because as the user continuously
+          // drags the resize bars, a stream of "Resizing" messages are
+          // sent to the window, and it would be pointless (and slow)
+          // to resize for each "Resizing" message received from dragging
+          // the resize bars.
+          //
+          // So instead, we reset after the user is done resizing the
+          // window and releases the resize bars, which sends a
+          // "WindowStopWindowResize" message.
+        }
+        else // API call such as SetWindowPos or mSwapChain->SetFullscreenState.
+        {
+          REX_WARN(LogWindows, "API call such as SetWindowPos or mSwapChain->SetFullscreenState will also invoke a resize event.");
+        }
+      }
+
+      void resize(const event_system::Event& /*evt*/)
+      {
+        // Resize window ( although we might want to capture this within the window itself ... )
+        //
+        // Flush Command Queue
+        //
+        // Release front and back buffer
+        // Release depth stencil buffer
+        //
+        // Reset current back buffer back to first entry ( aka 0 )
+        //
+        // Recreate swap chain buffers
```

Comment:
I think you only want to do this on end resize not during resizing.
you want to stop rendering on resize start and resume rendering on resize finish.
no rendering needs to happen during resizing